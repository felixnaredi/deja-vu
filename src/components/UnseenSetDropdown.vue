<template>
  <div>
    <select
      class="p-2 bg-white border-2 disabled:bg-slate-100"
      :disabled="disabled"
      @change="(event) => $emit('change', event)"
    >
      <option
        v-for="unseenSetID in unseenSetIDs"
        :key="unseenSetID.menuItemLabel"
        :value="unseenSetID.primitive"
        :selected="selected == unseenSetID.primitive"
      >
        {{ unseenSetID.menuItemLabel }}
      </option>
    </select>
  </div>
</template>

<script lang="ts">
import UnseenSetID from "@/service/unseen-set-id";

export default {
  computed: {
    unseenSetIDs: () => {
      return [UnseenSetID.Top999WiktionaryFr, UnseenSetID.DictionaryFr01];
    },
  },
  props: {
    disabled: Boolean,
    selected: Number,
  },
  emits: ["change"],
};
</script>
