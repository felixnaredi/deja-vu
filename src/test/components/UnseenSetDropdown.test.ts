import UnseenSetDropdown from "@/components/UnseenSetDropdown.vue";
import UnseenSetID from "@/service/unseen-set-id";
import { mount } from "@vue/test-utils";

const unseenSetIDs = [
  UnseenSetID.DictionaryFr01,
  UnseenSetID.Top999WiktionaryFr,
];

let wrapper = mount(UnseenSetDropdown);
beforeEach(() => {
  wrapper = mount(UnseenSetDropdown);
});

describe("unit tests", () => {
  test("UnseenSetDropdown has the expected data", () => {
    expect(wrapper.vm.unseenSetIDs.length).toBe(unseenSetIDs.length);
    for (const id of unseenSetIDs) {
      expect(wrapper.vm.unseenSetIDs).toContainEqual(id);
    }
  });
  test("UnseenSetDropdown contains expected menu items", () => {
    const menuItemLabels = unseenSetIDs.map((id) => id.menuItemLabel);
    for (const option of wrapper.findAll("option")) {
      expect(menuItemLabels).toContain(option.text());
    }
  });
  test("'change' is emitted when dropdown menu is changed", () => {
    wrapper.find("select").trigger("change");
    wrapper.vm.$nextTick();
    expect(wrapper.emitted().change).toBeTruthy();
  });
  test("disabled is false by default", () => {
    expect(wrapper.props("disabled")).not.toBeTruthy();
  });
  test("selected element affects the UnseenSetDropdown", () => {
    const ws = unseenSetIDs.map((id) =>
      mount(UnseenSetDropdown, { props: { selected: id.primitive } })
    );
    while (ws.length > 1) {
      expect(ws).toContain(ws[ws.length - 1]);
      const w = ws.pop();
      expect(ws).not.toContain(w);
    }
  });
});

describe("snapshot", () => {
  test("snapshot matches with default UnseenSetDropdown", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
