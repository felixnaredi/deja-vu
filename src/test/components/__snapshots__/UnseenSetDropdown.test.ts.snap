// Vitest Snapshot v1

exports[`snapshot > snapshot matches with default UnseenSetDropdown 1`] = `
"<div><select class=\\"p-2 bg-white border-2 disabled:bg-slate-100\\">
    <option value=\\"2\\">ğŸ‡«ğŸ‡· 999 mots les plus courants</option>
    <option value=\\"1\\">ğŸ‡«ğŸ‡· liste franÃ§ais</option>
  </select></div>"
`;
